build-aux/com.obsproject.Studio.json

```diff
@@ -23,7 +23,8 @@
         "--env=LADSPA_PATH=/app/extensions/Plugins/ladspa",
         "--env=LV2_PATH=/app/extensions/Plugins/lv2:/app/lib/lv2",
         "--env=VST_PATH=/app/extensions/Plugins/vst",
-        "--env=VST3_PATH=/app/extensions/Plugins/vst3"
+        "--env=VST3_PATH=/app/extensions/Plugins/vst3",
+        "--env=ENGINE_OPTION_PATH_BINARIES=/app/bin"
```

Comment:
`$PATH` already is set to `/app/bin:/usr/bin`, or even `CMAKE_INSTALL_PREFIX` is set to `/app` (default is `/usr`).

Is there really a need to has such environment variable ?

Reply:
this is the carla path, which typically is /usr/lib/carla/ but that breaks under flatpak, so we need to force things here

Reply:
So normally Carla CMake get `CMAKE_INSTALL_PREFIX` set to `/app`, so it's libdir became automatically `/app/lib` if you rely on CMake function and variable.

Reply:
> this is the carla path, which typically is /usr/lib/carla/ but that breaks under flatpak, so we need to force things here

Okay, I get we are more in a libexec situation than a bin one. But this customization `CARLA_SKIP_LIBDIR_SUFFIX` and `ENGINE_OPTION_PATH_BINARIES` really looks like (I didn't check your code this is an assumption) you don't rely on CMake for getting the libdir.

Reply:
Carla relies on its libs, executables and utils to be all in the same dir. it is typically /usr/lib/carla on Linux (and no, it can't be separated as this mixes regular libraries, executables, windows bridges and winelib stuff; splitting these would be too much hassle, so a single directory under a custom suffix is chosen).

Somehow the rpath set by obs carla-bridge is being stripped away during install, leading to obs-bridge.so not being able to find libcarla_utils.so. My current solution was to simply not use the lib suffix, since flatpak is a self-contained system it does need the separation as we can easily deal with any conflicts that arise.

Shipping with the bridge tools in $prefix/lib is wrong though, so they go into $prefix/bin.
This breaks Carla's expectations of its libs being on the same dir as the tools though. So we nudge things a bit to get it all correct again.

Reply:
> Shipping with the bridge tools in $prefix/lib is wrong though, so they go into $prefix/bin.

This is the concept of libexec, (e.g. Fedora put such binaries them in `/usr/libexec` and ArchLinxu in `/usr/lib`.
If the binary is not meant to be used by the user but mainly by libraries of services, they can go to libexec.

Reply:
they can be, the tools are useful by itself. I use them quite often when needing to quickly check if a plugin binary is okay, working as a sort of validation tool. I also use them in automated CI tests to check for memory leaks and other issues.
so they could even be in /usr/bin by default, it is just that they would then be separated from other tools (like carla-bridge-win64.exe and jackbridge64.dll for example) which would  make it a mess to manage.

Reply:
Okay, so bin or libexec at the end `ENGINE_OPTION_PATH_BINARIES` should not be needed since the binaries should be aware of the prefix used `/app` and with `CARLA_SKIP_LIBDIR_SUFFIX` aware that it's in the bin directory under the prefix.

Reply:
not really, the `CARLA_SKIP_LIBDIR_SUFFIX` option just changes where the libraries will be installed to, it does not change the build.
Carla doesnt really have a concept of "PREFIX" or "LIBDIR", finding the tools is based on where the main lib lives, so we could move /usr/lib/carla into some random /opt/customapp/lib/carla and everything would still work as usual.

preferably I would fix the rpath issue, but I dont know enough of flatpak to be able to even start to debug this.

Reply:
> they can be, the tools are useful by itself. I use them quite often when needing to quickly check if a plugin binary is okay, working as a sort of validation tool. I also use them in automated CI tests to check for memory leaks and other issues.
> so they could even be in /usr/bin by default, it is just that they would then be separated from other tools (like carla-bridge-win64.exe and jackbridge64.dll for example) which would make it a mess to manage.

As the author of the PR it's absolutely in your purview to decide the scope of the change you want to add to OBS Studio, but I want to carefully caution that the bigger the scope of the change, the harder it will be to get it approved and merged.

I.e., there is no need to have separate binaries for utility, so those would be beyond the desired scope. Dropping those entirely would resolve the question of where to put them and thus move the plugin closer to a possible finish line.

Reply:
as I mentioned somewhere else, there is 100% the need for separate binaries, that is how process separation works in the first place, we need to run *something* in a separate process and that requires having a tool that we can run.

for this particular case I found an alternative, I have some pending changes to push to the PR but want to finalize them in one-go in order to minimize the constant stream of changes (it also needs a version bump on obs-deps)